### PR TITLE
feat!: Refactor RowId/SampleId/NamespacedDataId related API

### PR DESCRIPTION
**Pull request by oblique**

### CODE REVIEW

File: types/src/data_availability_header.rs [hunk 1]
```diff
@@ -53,22 +53,32 @@ use crate::{bail_validation, Error, ExtendedDataSquare, Result, ValidateBasic, V
...
     /// Merkle roots of the [`ExtendedDataSquare`] columns.
-    pub column_roots: Vec<NamespacedHash>,
+    column_roots: Vec<NamespacedHash>,
 }
 
 impl DataAvailabilityHeader {
+    /// Create new [`DataAvailabilityHeader`].
+    ///
+    /// [`DataAvailabilityHeader::validate_basic`] should be called after this to check validity.
+    pub fn new(row_roots: Vec<NamespacedHash>, column_roots: Vec<NamespacedHash>) -> Self {
```
Comment (fl0rek) [inline types/src/data_availability_header.rs:65]: if this can create invalid DAH, maybe `new_unchecked`?

Comment (fl0rek) [reply types/src/data_availability_header.rs:65]: Or make this new checked and add an extension trait with unchecked version for testing, but that feels like overkill.

**Review by fl0rek — COMMENTED** [review]
Looks much neater now 🙌 
left few small comments

File: types/src/data_availability_header.rs (hunk 1 above)
Comment (oblique) [reply types/src/data_availability_header.rs:65]: I'm ok on having both actually.

Comment (oblique) [reply types/src/data_availability_header.rs]: Added both, what do you think?

File: node/src/daser.rs [hunk 2]
```diff
@@ -179,18 +178,24 @@ where
     }
 }
 
-fn random_indexes(block_len: usize, max_samples_needed: usize) -> HashSet<usize> {
+fn random_indexes(square_width: u16, max_samples_needed: usize) -> HashSet<(u16, u16)> {
+    let block_len = usize::from(square_width) * usize::from(square_width);
```
[the remark this answers is not in the document]
Comment (fl0rek) [reply node/src/daser.rs]: eh, names are always the hardest part 🙂 
maybe `max_samples_in_block`/`square`? But I wont cry too much, if it's left as it is.

**Review by fl0rek — APPROVED** [review]
🎸

**Review by zvolin — APPROVED** [review]
nice

**Review by fl0rek — APPROVED** [review]
lgtm!

**Pull request merged**